dotCMS/src/main/java/com/dotcms/util/VanityUrlUtil.java

```diff
@@ -0,0 +1,54 @@
+package com.dotcms.util;
+
+import com.dotcms.contenttype.model.type.VanityUrlContentType;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+
+/**
+ * This class provide some utility methods to interact with
+ * Vanity URLs
+ * @author oswaldogallango
+ *
+ */
+public class VanityUrlUtil {
+
+	/**
+	 * Generate the sanitized cache key
+	 * @param key to be used in the cache
+	 * @param languageId 
+	 * @return String with the sanitized key name
+	 */
+	public static String sanitizeKey(String key, long languageId){
```

Comment:
this method was changed, now receive host, uri and languageid, the only one not final is the uri because we update the value with a fixUri method